### PR TITLE
Adds APIs for listing project and branch names in Unison Local

### PR DESCRIPTION
**Pull request by ChrisPenner**
Depends on #4246 

## Overview

We'd like to have a project-branch switcher on local UI just like Share does. This provides the necessary APIs to build such a thing.

Adds the following new endpoints mounted under `/api/*` to the unison-local API.

```json
GET /projects
[
    {
        "projectName": "argparse"
    },
    {
        "projectName": "free-applicative"
    }
]
```

```json
GET /projects/<project-name>/branches?prefix=<prefix>
[
    {
        "branchName": "main"
    },
    {
        "branchName": "@runarorama/main"
    }
]
```

## Implementation notes

* Remove all the old "project" apis which were used for listing "project" for the OLD ucm-based share server. (Simone said this was okay now)
* Add the new APIs for listing Projects and Branches.
* Tweak queries to use Sqlite Globs
* Add the new handlers using existing queries.

## Test coverage

New transcript

## Loose ends

Currently this ONLY returns the name, in the future we'll probably want a bit more info here, but this should be enough to bootstrap the frontend from :)

### CODE REVIEW

File: codebase2/codebase-sqlite/U/Codebase/Sqlite/Queries.hs [hunk 1]
```diff
@@ -3265,21 +3265,16 @@ loadAllProjects =
     |]
 
 -- | Load all projects whose name matches a prefix.
-loadAllProjectsBeginningWith :: Text -> Transaction [Project]
-loadAllProjectsBeginningWith prefix =
-  -- since we are not likely to many projects, we just get them all and filter in Haskell. This seems much simpler than
-  -- running a LIKE query, and dealing with escaping, case sensitivity, etc
```
Comment (ChrisPenner) [inline codebase2/codebase-sqlite/U/Codebase/Sqlite/Queries.hs:3271]: I'm guessing maybe it just wasn't super clear how to use `globEscape` when this was written, but it's really not tricky to use SQLite for this so I swapped it in.

GLOB is case-sensitive, which matches the existing `Text.isPrefixOf` behaviour.

SQLite is smart enough to optimize it to a prefix search, or to ignore it entirely if the glob is `*`.

File: unison-share-api/src/Unison/Server/Local/Endpoints/Projects.hs
Comment (ChrisPenner) [inline unison-share-api/src/Unison/Server/Local/Endpoints/Projects.hs:1]: You can ignore the ENTIRE left-hand-side of this file's diff, it's the old UCM-based Share project implementation which is now completely defunct.

This is just a good module to put the new stuff in, but it's all new.

**Review by hojberg — APPROVED** [review]
Nice!

**Comment by aryairani** [comment]
It looks like it needs updated transcript outputs committed

**Comment by ChrisPenner** [comment]
> It looks like it needs updated transcript outputs committed

Ah, yeah looks like I forgot to delete the old project api ones.

P.s. reminder that you can feel free to still approve or request changes pending fixing a simple CI error like this, might save you a back-and-forth looking at PRs again later. Up to you though :)

**Review by aryairani — APPROVED** [review]

**Pull request merged**